src/text_engine/boundaries/Room.java

```diff
@@ -47,12 +47,11 @@ public Room(@NotNull String name, @NotNull String description,
             throws IllegalArgumentException {
         super(name, description);
 
-        if (items.size() > CONTENT_LIMIT) {
-            throw new IllegalArgumentException(String.format("Rooms can have a maximum of %d items.",
-                                                             CONTENT_LIMIT));
+        this.items = new ArrayList<>(items.size());
+        for (Item i : items) {
+            addItem(i);
```

Comment:
Did this to reduce redundancy with checking `items.size()` against `CONTENT_LIMIT`. `addItem(i)` will throw an exception if the content limit is reached.